src/config/config.ts

```diff
@@ -184,68 +149,70 @@ export class Config extends ConfigFile<ConfigFile.Options> {
       Config.messages = Messages.loadMessages('@salesforce/core', 'config');
     }
 
-    if (!Config.allowedProperties) {
-      Config.allowedProperties = [
-        {
-          key: 'instanceUrl',
-          input: {
-            // If a value is provided validate it otherwise no value is unset.
-            validator: value => value == null || (isString(value) && sfdc.isSalesforceDomain(value)),
-            failedMessage: Config.messages.getMessage('InvalidInstanceUrl')
-          }
-        },
-        {
-          key: Config.API_VERSION,
-          hidden: true,
-          input: {
-            // If a value is provided validate it otherwise no value is unset.
-            validator: value => value == null || (isString(value) && sfdc.validateApiVersion(value)),
-            failedMessage: Config.messages.getMessage('InvalidApiVersion')
-          }
-        },
-        { key: Config.DEFAULT_DEV_HUB_USERNAME },
-        { key: Config.DEFAULT_USERNAME },
-        {
-          key: Config.ISV_DEBUGGER_SID,
-          encrypted: true,
-          input: {
-            // If a value is provided validate it otherwise no value is unset.
-            validator: value => value == null || isString(value),
-            failedMessage: Config.messages.getMessage('InvalidIsvDebuggerSid')
-          }
-        },
-        {
-          key: Config.ISV_DEBUGGER_URL,
-          input: {
-            // If a value is provided validate it otherwise no value is unset.
-            validator: value => value == null || isString(value),
-            failedMessage: Config.messages.getMessage('InvalidIsvDebuggerUrl')
-          }
-        },
-        {
-          key: Config.DISABLE_TELEMETRY,
-          input: {
-            validator: value => value == null || ['true', 'false'].includes(value.toString()),
-            failedMessage: Config.messages.getMessage('InvalidBooleanConfigValue')
-          }
-        },
-        // This should be brought in by a plugin, but there isn't a way to do that right now.
-        {
-          key: 'restDeploy',
-          hidden: true,
-          input: {
-            validator: value => value != null && ['true', 'false'].includes(value.toString()),
-            failedMessage: Config.messages.getMessage('InvalidBooleanConfigValue')
-          }
-        },
-        {
-          key: Config.MAX_QUERY_LIMIT,
-          input: {
-            validator: value => isNumber(value),
-            failedMessage: Config.messages.getMessage('InvalidNumberConfigValue')
-          }
+    Config.allowedProperties = [
```

Comment:
It seems like we can run into problems when resetting this every time since this class doesn't cache. i.e. a plugin adds some config and another plugin creates one of these manually, will it clear out the other plugins properties?